destination_sdk_v2.proto

```diff
@@ -0,0 +1,181 @@
+    rpc Test (TestRequest) returns (TestResponse) {}
+    rpc DescribeTable (DescribeTableRequest) returns (DescribeTableResponse) {}
+    rpc CreateTable(CreateTableRequest) returns (CreateTableResponse) {}
+    rpc AlterTable(AlterTableRequest) returns (AlterTableResponse) {}
+    rpc Truncate(TruncateRequest) returns (TruncateResponse) {}
+    rpc WriteBatch (WriteBatchRequest) returns (WriteBatchResponse) {}
+}
+
+message CapabilitiesRequest {}
+
+message CapabilitiesResponse {
+    BatchFileFormat batch_file_format = 1;
+    repeated DataTypeMappingEntry data_type_mappings = 2;
+    bool supports_history_mode = 3;
+}
+
+message DataTypeMappingEntry {
+    DataType fivetran_type = 1;
+    oneof destination_type {
+        DestinationType map_to = 2;
+        bool unsupported = 3;
+    }
+}
+
+message DestinationType {
+    string name = 1;
+    oneof mapping {
+        bool keep_same = 2;
+        DataType map_to = 3;
+    }
+    optional MaxValue max_value = 4;
+}
+
+message MaxValue {
+    oneof max_param {
+        int32 numeric_param = 1;
+        DecimalParams decimal_param = 2;
+        google.protobuf.Timestamp date_param = 3;
+    }
+}
+
+message DescribeTableRequest {
+    map<string, string> configuration = 1;
+    string schema_name = 2;
+    string table_name = 3;
+}
+
+message DescribeTableResponse {
+    oneof response {
+        bool not_found = 1;
+        string failure = 2;
+        Table table = 3;
+    }
+}
+
+message CreateTableRequest {
+    map<string, string> configuration = 1;
+    string schema_name = 2;
+    Table table = 3;
+}
+
+message CreateTableResponse {
+    oneof response {
+        bool success = 1;
+        string failure = 2;
+    }
+}
+
+message AlterTableRequest {
+    map<string, string> configuration = 1;
+    string schema_name = 2;
+    string table_name = 3;
+    repeated SchemaDiff changes = 4;
+}
+
+message SchemaDiff {
+    oneof change {
+        Column add_column = 1;
+        ChangeType change_column_type = 2;
+        PrimaryKey updated_primary_keys = 3;
+    }
+}
+
+message PrimaryKey {
+    repeated string column_name = 1;
+}
+
+message ChangeType {
+    string column_name = 1;
+    DataType new_type = 2;
+    optional DataTypeParams params = 3;
+}
+
+message AlterTableResponse {
+    oneof response {
+        bool success = 1;
+        string failure = 2;
+    }
+}
+
+message TruncateRequest {
+    map<string, string> configuration = 1;
+    string schema_name = 2;
+    string table_name = 3;
+    string synced_column = 4;
+    google.protobuf.Timestamp utc_delete_before = 5;
+    optional SoftTruncate soft = 6;
+}
+
+message SoftTruncate {
+    string deleted_column = 3;
+}
+
+message TruncateResponse {
+    oneof response {
+        bool success = 1;
+        string failure = 2;
+    }
+}
+
+message WriteBatchRequest {
+    map<string, string> configuration = 1;
+    string schema_name = 2;
+    Table table = 3;
+    bool history_mode = 4;
```

Comment:
I think we have a change for this field. If I remember correctly, we are replacing it with a dedicated message type. Please talk to @fivetran-abdulsalam about it.

Reply:
Yes, it is this change - https://github.com/fivetran/engineering/pull/180958/files#diff-0402336e684efdd8324b684e67c68aab81031f1c13987bab3e1b3504f389a01dR149-R152.